.github/actions/e2e/cleanup-nap/action.yaml

```diff
@@ -0,0 +1,48 @@
+name: Cleanup
+description: 'Runs all the cleanup tasks to cleanup resources deployed during E2E'
+inputs:
+  client-id:
+    description: "Client ID"
+    required: true
+  tenant-id:
+    description: "Tenant ID"
+    required: true
+  subscription-id:
+    description: "Subscription ID"
+    required: true
+  # region:
+  #   description: "Region to create aks cluster"
+  #   required: true
+  resource_group:
+    description: "Name of the resource group to create the cluster within"
+    required: true
+  cluster_name:
+    description: 'Name of the cluster to be created'
+    required: true
+  acr_name:
+    description: "Name of the acr holding the karpenter image"
+    required: true
+  git_ref:
+    description: "The git commit, tag, or branch to check out"
+    required: false
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      with:
+        ref: ${{ inputs.git_ref }}
+    - name: az login
+      uses: azure/login@cb79c773a3cfa27f31f25eb3f677781210c9ce3d # v1.6.1
+      with:
+        client-id: ${{ inputs.client-id }}
+        tenant-id: ${{ inputs.tenant-id }}
+        subscription-id: ${{ inputs.subscription-id }}
+    - name: az set sub
+      shell: bash
+      run: az account set --subscription ${{ inputs.subscription-id }}
+    - name: delete cluster ${{ inputs.cluster_name }}
+      shell: bash
+      run: az aks delete --name ${{ inputs.cluster_name }} --resource-group ${{ inputs.resource_group }} --yes --no-wait
```

Comment:
Just doesn't have the delete `acr`? set. Could simply remove that from the general cleanup since resource group delete should be enough. I think that's worth removing the need for an almost duplicate file.